android/java/org/chromium/chrome/browser/BraveActivity.java

```diff
@@ -239,13 +241,18 @@ private void createNotificationChannel() {
     }
 
     private void setupBraveSetDefaultBrowserNotification() {
-        Context context = ContextUtils.getApplicationContext();
-        if (BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(this)) {
-            // Don't ask again
-            return;
-        }
-        Intent intent = new Intent(context, BraveSetDefaultBrowserNotificationService.class);
-        context.sendBroadcast(intent);
+        // Post task to IO thread because isBraveSetAsDefaultBrowser may cause
+        // sqlite file IO operation underneath
+        PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK, () ->
```

Comment:
this is nice thing to know.

Reply:
Yeah, it was a surprise for me. I saw it on Samsung Android 8 device but didn't on Xiaomi Android 9 device.
here is a part of logs from Samsung 
```
04-22 15:40:02.737: D/StrictMode(11681): StrictMode policy violation; ~duration=144 ms: android.os.StrictMode$StrictModeDiskReadViolation: policy=22085695 violation=2
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1438)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteConnection.applyBlockGuardPolicy(SQLiteConnection.java:1247)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteConnection.executeForCursorWindow(SQLiteConnection.java:1014)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:836)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:152)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:141)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:219)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:258)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.storage.EdmStorageProviderBase.getValues(EdmStorageProviderBase.java:389)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.queryAllDefaultAppIntents(ApplicationPolicy.java:12763)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.retrieveDefaultAppFromDb(ApplicationPolicy.java:12802)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.enterprise.application.ApplicationPolicy.getDefaultApplicationInternal(ApplicationPolicy.java:12109)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.filterCandidatesWithDomainPreferredActivitiesLPr(PackageManagerService.java:10180)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.queryIntentActivitiesInternal(PackageManagerService.java:9838)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.resolveIntentInternal(PackageManagerService.java:8772)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.resolveIntent(PackageManagerService.java:8745)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:930)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:5096)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Binder.execTransact(Binder.java:682)
04-22 15:40:02.737: D/StrictMode(11681): # via Binder call with stack:
04-22 15:40:02.737: D/StrictMode(11681): android.os.StrictMode$LogStackTrace
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.StrictMode.readAndHandleBinderCallViolations(StrictMode.java:2137)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Parcel.readExceptionCode(Parcel.java:1941)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Parcel.readException(Parcel.java:1910)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.content.pm.IPackageManager$Stub$Proxy.resolveIntent(IPackageManager.java:4554)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ApplicationPackageManager.resolveActivityAsUser(ApplicationPackageManager.java:1035)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ApplicationPackageManager.resolveActivity(ApplicationPackageManager.java:1029)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.notifications.BraveSetDefaultBrowserNotificationService.isBraveSetAsDefaultBrowser(BraveSetDefaultBrowserNotificationService.
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.BraveActivity.setupBraveSetDefaultBrowserNotification(BraveActivity.java:249)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.BraveActivity.performPostInflationStartup(BraveActivity.java:167)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeTabbedActivity.performPostInflationStartup(ChromeTabbedActivity.java:1547)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.postInflationStartup(AsyncInitializationActivity.java:182)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.onInflationComplete(ChromeBrowserInitializer.java:176)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.lambda$handlePreNativeStartup$0$ChromeBrowserInitializer(ChromeBrowserInitializer.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer$$Lambda$0.run(ChromeBrowserInitializer.java:0)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onInitialLayoutInflationComplete(AsyncInitializationActivity.java:753)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.onInitialLayoutInflationComplete(ChromeActivity.java:596)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.doLayoutInflation(ChromeActivity.java:568)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.ChromeActivity.triggerLayoutInflation(ChromeActivity.java:522)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.setContentViewAndLoadLibrary(AsyncInitializationActivity.java:168)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.ChromeBrowserInitializer.handlePreNativeStartup(ChromeBrowserInitializer.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onCreateInternal(AsyncInitializationActivity.java:318)
04-22 15:40:02.737: D/StrictMode(11681): <----->at org.chromium.chrome.browser.init.AsyncInitializationActivity.onCreate(AsyncInitializationActivity.java:264)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.Activity.performCreate(Activity.java:7183)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1220)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2910)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3032)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.-wrap11(ActivityThread.java:0)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Handler.dispatchMessage(Handler.java:105)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.os.Looper.loop(Looper.java:164)
04-22 15:40:02.737: D/StrictMode(11681): <----->at android.app.ActivityThread.main(ActivityThread.java:6940)
04-22 15:40:02.737: D/StrictMode(11681): <----->at java.lang.reflect.Method.invoke(Method.java)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
04-22 15:40:02.737: D/StrictMode(11681): <----->at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)

```